soh/src/code/z_message_RandoNaviTips.c

```diff
@@ -0,0 +1,313 @@
+#include "global.h"
+#include "message_data_static.h"
+#include "vt.h"
+
+#include <string.h>
+
+// Function for letting Navi give general rando tips to the player instead of her
+// normal generic quest tips.
+
+// Only applies to Navi's overworld quest hints - does not apply to enemy information
+// or contextual hints (e.g. "This hallway is twisted!" in Forest Temple)
+
+void RandoNaviTip(GlobalContext* globalCtx) {
+    u16 randNaviTip = rand() % 15; // Random number between 0 and total number of Navi tips
+    MessageContext* msgCtx = &globalCtx->msgCtx;
+    Font* font = &msgCtx->font;
+
+    if (randNaviTip == 0) {
+        switch (gSaveContext.language) {
+            case LANGUAGE_FRA:
+                strcpy(font->msgBuf, "\x05\x44"
+                                     "French tip about playing rando! 0\x02");
+                break;
+            case LANGUAGE_GER:
+                strcpy(font->msgBuf, "\x05\x44"
+                                     "German tip about playing rando! 0\x02");
+                break;
+            case LANGUAGE_ENG:
+            default:
+                strcpy(font->msgBuf, "\x05\x44"
+                                     "Missing a small key in a dungeon?\x01"
+                                     "Maybe the\x05\x41 boss\x05\x44 has it!\x02");
+                break;
+        }
+    }
```

Comment:
For the custom-message sytem, you can create a message table and register the messages in a function in `randomizer.cpp` or a new adjacent file, and call that function in InitRando(). You can probably use the `CreateScrubMessages` function as an example. Then you can use the `randoNaviTip` values here as the `textID`s and just run `CustomMessageManager::Instance->RetrieveMessage(<your-table-id>, randoNaviTip)`. Set that equal to a value and have this function return it. `CustomMessage_RetrieveIfExists` handles copying the message into the buffer for you.